### PR TITLE
Add tests for Annex B extn: O.p.__proto__

### PR DESCRIPTION
**Pull request by jugglinmike**

### CODE REVIEW

File: test/annexB/built-ins/Object/prototype/__proto__/set-non-object.js [hunk 1]
```diff
@@ -0,0 +1,18 @@
+// Copyright (C) 2016 the V8 project authors. All rights reserved.
+// This code is governed by the BSD license found in the LICENSE file.
+/*---
+esid: sec-object.prototype.__proto__
+es6id: B.2.2.1
+description: Called on a value that is object-coercible but not an Object
+info: >
+    1. Let O be ? RequireObjectCoercible(this value).
+    2. If Type(proto) is neither Object nor Null, return undefined.
```
Comment (leobalter) [inline test/annexB/built-ins/Object/prototype/__proto__/set-non-object.js]: This undefined return is from step 3

```
3. If Type(O) is not Object, return undefined.
```

File: test/annexB/built-ins/Object/prototype/__proto__/set-immutable.js [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+    [...]
+    4. Let status be ? O.[[SetPrototypeOf]](proto).
+    5. If status is false, throw a TypeError exception.
+---*/
+
+Object.prototype.__proto__ = null;
+
+assert.throws(TypeError, function() {
+  Object.prototype.__proto__ = {};
+});
```
Comment (leobalter) [inline test/annexB/built-ins/Object/prototype/__proto__/set-immutable.js]: this is so tricky

**Comment by leobalter** [comment]
One comment above to fix that doesn't affect the tests behavior.

LGTM

**Comment by jugglinmike** [comment]
I've extended the meta data of `set-immutable.js` as requested.

**Pull request merged**